provider/aws/aws.go

```diff
@@ -637,6 +637,10 @@ func (p *AWSProvider) createUpdateChanges(newEndpoints, oldEndpoints []*endpoint
 	var updates []*endpoint.Endpoint
 
 	for i, newE := range newEndpoints {
+		if i >= len(oldEndpoints) || oldEndpoints[i] == nil {
+			log.Debugf("skip %s as endpoint not found in current endpoints", newE.DNSName)
+			continue
+		}
 		oldE := oldEndpoints[i]
```

Comment:
```suggestion
		oldE := oldEndpoints[i]
		if i >= len(oldEndpoints) || oldE == nil {
			log.Debugf("skip %s as endpoint not found in current endpoints", newE.DNSName)
			continue
		}
```

Reply:
Doing oldE := oldEndpoints[i] before check is a potential nill pointer and index out of range. Let's have a look if test fail

Reply:
I've updated test to show case index out of range
<img width="1384" height="830" alt="Screenshot 2025-07-12 at 10 08 25" src="https://github.com/user-attachments/assets/dad11005-cd14-49bd-b549-e80520d2dcbe" />

Reply:
\o that's right. Thanks !